### PR TITLE
Introduce a config option for station mode

### PR DESCRIPTION
**Pull request by rado17**
Introduce a config option for station mode which can be used to include station-only features during compile time.

### CODE REVIEW

**Review by sachinthegreen — CHANGES_REQUESTED** [review]
Can we include the Jira reference in the commits please?

File: modules/hostap/Kconfig [hunk 1]
```diff
@@ -156,6 +156,11 @@ config WPA_SUPP_WPA3
 	depends on !WPA_SUPP_CRYPTO_NONE
 	default y
 
+config WPA_SUPP_STA
+	bool "Station mode support"
+	default y
+	depends on !WPA_SUPP_AP
```
[the remark this answers is not in the document]
Comment (rado17) [reply modules/hostap/Kconfig:162]: We can add a config option to (say `ENABLE_DHCP_CLIENT`), and define it only for STA mode at build time. WDYT?

[the remark this answers is not in the document]
Comment (krish2718) [reply modules/hostap/Kconfig:162]: That again brings in the question, how do you do STA mode in WPA supp? Anyways, and in upstream we are talking abut concurrent STA + AP modes as well. So, yes, the solution should be per-iface DHCP Client/Static IP/DHCP server option.

[the remark this answers is not in the document]
Comment (rado17) [reply modules/hostap/Kconfig]: Added checks specific to interface acting as STA.

**Review by sachinthegreen — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

**Comment by krish2718** [comment]
manifest is updated in https://github.com/nrfconnect/sdk-nrf/pull/15810

**Pull request closed without merging**